src/resources/Transfers/types.ts

```diff
@@ -0,0 +1,150 @@
+import { ISODateString } from '../../utils/types'
+
+type TransferType = 'URL' | 'TORRENT' | 'PLAYLIST' | 'LIVE_STREAM'
+
+type TransferStatus =
+  | 'WAITING'
+  | 'IN_QUEUE'
+  | 'DOWNLOADING'
+  | 'COMPLETING'
+  | 'STOPPING'
+  | 'SEEDING'
+  | 'COMPLETED'
+  | 'ERROR'
+  | 'PREPARING_SEED'
+
+type BaseTransfer = {
+  availability: null | number
+  created_at: ISODateString
+  current_ratio: null | number
+  down_speed: number
+  downloaded: number
+  error_message: null | string
+  estimated_time: null | number
+  file_id: null | number
+  finished_at: null | ISODateString
+  id: number
+  is_private: boolean
+  name: string
+  save_parent_id: number
+  seconds_seeding: null | number
+  size: number
+  source: string
+  started_at: null | ISODateString
+  status: TransferStatus
+  type: TransferType
+  subscription_id: null | number
+  uploaded: number
+  up_speed: number
+
+  // not documented in swagger
+  callback_url: null | string
+  client_ip: null | string
+  completion_percent: number // used?
+  created_torrent: boolean // used?
+  download_id: number
+  hash: null | string // used?
```

Comment:
when does it transition to `string` from `null`?

Reply:
Torrent ise.